### ISSUE TITLE
Ability to Compile Dart Code and Distribute it as libraries

### ISSUE THREAD

**Issue by canewsin** [dart-lang/sdk#37952]
Every Code distributed as library is visible for user, like c/c++ if there is an ability to distribute as bin libraries is better feature

**Comment by munificent** [comment]
Moving this to the SDK repo because I think it's *slightly* more of an implementation concern than a language design issue.

**Comment by ZeekHuge** [comment]
Or is their a way to dart code as minifed and somewhat obfuscated format ? The goal is to be able to use this dart code as a dependency in a flutter application, but without exposing all the source code in an extremely easy to read format.

**Comment by Defuera** [comment]
Is there any update on the matter?

**Comment by mncmeditech** [comment]
The agora_rtc_engine seems to be doing it. Any insight on how the package https://pub.dev/packages/agora_rtc_engine is able to implement it - hide the source code, yet allow users to use it?

**Comment by ZeekHuge** [comment]
@munificent Not really. All they have done is, not share the URL for the source on the pub. Once you add it as a dependency, all the code, in that package at-least, can be browsed. And even the source can be download from https://pub.dev/packages/agora_rtc_engine/versions

**Comment by ZeekHuge** [comment]
@mncmeditech Not really. All they have done is, not share the URL for the source on the pub. Once you add it as a dependency, all the code, in that package at-least, can be browsed. And even the source can be download from https://pub.dev/packages/agora_rtc_engine/versions

**Comment by mncmeditech** [comment]
Thanks a lot for the clarification.
